modules/neo4j/src/main/java/org/testcontainers/containers/Neo4jContainer.java

```diff
@@ -230,6 +231,9 @@ public S withEnterpriseEdition() {
      * @return This container.
      */
     public S withAdminPassword(final String adminPassword) {
+        if (adminPassword != null && adminPassword.length() < 8) {
+            logger().warn("Your provided admin password is too short and will not work with Neo4j 5.3+.");
```

Comment:
I think this warning here is better than a tighter integration with more assumptions, that requires knowledge about the used Neo4j version 👍